framework/CodeInterfaces/PHISICS/phisicsdata.py

```diff
@@ -0,0 +1,961 @@
+        for line in outfile:
+          line = self.removeSpaces(line)
+          if re.search(r'TIME',line):
+            line = re.sub(r'\n',r'',line)
+            self.isotopeList = line.split(',')
+          if re.search(r'Material',line):
+            materialList = line.split(',')
+            matList = self.getMaterialList(line.split(','), matList)
+          stringIsFloatNumber = self.isFloatNumber(line.split(','))
+          if stringIsFloatNumber is True:
+            line = re.sub(r'\n',r'',line)
+            if (float(line.split(',')[0]) == float(matchedTimeSteps[timeStepIndex])):
+              for i in xrange (1,len(self.isotopeList)):
+                timeStepList.append(line.split(',')[0])
+                depLabelList.append('dep'+'|'+materialList[1]+'|'+self.isotopeList[i])
+                depList.append(line.split(',')[i])
+    return depLabelList, depList, timeStepList, matList
+  
+  def getDepInfoMrtau(self,timeStepIndex,matchedTimeSteps):
+    """
+      Reads the MRTAU csv file to get the material density info relative to depletion.
+      @ In, timeStepIndex, integer, timestep number
+      @ In, matchedTimeSteps, list, list of time steps considered
+      @ Out, depLabelList, depList
+    """
+    depList = []
+    with open(os.path.join(self.workingDir,self.mrtauOutputFileMPI[0]), 'r') as outfile:
+      for line in outfile:
+        line = self.removeSpaces(line)
+        stringIsFloatNumber = self.isFloatNumber(line.split(','))
+        if stringIsFloatNumber is True:
+          if (float(line.split(',')[0]) == float(matchedTimeSteps[timeStepIndex])):
+            line = re.sub(r'\n',r'',line)
+            for i in xrange (0,len(self.isotopeListMrtau)):
+              depList.append(line.split(',')[i+1])
+    return depList
+    
+  def findDecayHeat(self,line):
+    """
+      Determines whether the decay heat is printed.
+      @ In, line, string
+      @ Out, findDecayHeat, boolean, True if decay heat is printed, false otherwise
+      @ Out, DecayHeatUnits, string
+    """
+    DecayHeatUnits = None
+    isDecayHeatPrinted = line.split(' ')[-2]
+    if isDecayHeatPrinted == 'YES':
+      DecayHeatUnits =  line.split(' ')[-1]
+      return True, DecayHeatUnits
+    else:
+      return False, DecayHeatUnits
+  
+  def numberOfMediaUsed(self,mpi,numberOfMPI):
+    """
+      Finds the number of media used in a given INSTANT output.
+      @ In, mpi, integer, MPI number considered
+      @ In, numberOfMPI, integer, total number of MPI user-selected
+      @ Out, mediaUsed, integer, number of media treated in one INSTANT MPI output
+    """
+    count = 0
+    if self.phisicsRelap is True and mpi >= numberOfMPI - 1:
+      pass
+    else:
+      with open(os.path.join(self.workingDir, self.instantOutputFileMPI[mpi])) as outfile:
+        for line in outfile:
+          if re.search(r'Medium\s+\d+\s+used',line):
+            count = count + 1
```

Comment:
hum I am not sure, I tested your line it errors out.  This would run:
`outfile = open(os.path.join(self.workingDir, self.instantOutputFileMPI[mpi]))`
`test = next (((i,line) for i,line in enumerate(outfile.readlines()) if re.search(r'Medium\s+\d+\s+used',line)),None)`

but this return the line number where the medium is printed, while I just want to count how many medium are used